target/linux/brcm63xx/image/bcm63xx.mk

```diff
@@ -215,6 +215,17 @@ define Device/adb_a4001n1
 endef
 TARGET_DEVICES += adb_a4001n1
 
+define Device/adb_a4001n-a-000-1a1-ax
```

Comment:
I like that much more, but if we go the way of terribly long (but accurate) names anyway, the pdg should be included as well (pdg or p-dg).

Despite, if I understand you correctly, the current a4001n existing in OpenWrt is not precisely named at the moment, as it's actually a subspecies of the a4001n. Since I just recently changed the target names in brcm63xx anyway, I'm tempted to _correct_ that right away. Thoughts? And what should happen to a4001n1 then?

Reply:
I understand your point: it was the reason why I originally called it AGPWI in the files instead of its long ADB name.

The guy who added support for the ADB P.DG A4001N A-000-1A1-AE called the image "A4001N" for this reason too: for shortness. He most probably thought that "A4001N" could be more than enought to univocally identify this PCB, but it was not true.

To answer your question yes, "A4001N" should be corrected in "ADB P.DG A4001N A-000-1A1-AE". I was planning to do a PR for that. The same thing as to be done for the "A4001N1", this is not its full name but I did not search yet its full name.

I'm pretty sure that "P.DG" is just the full name of ADB company, not something userfull to univocally identify the PCB like "A-000-1A1-AE". I would not add this name too for shortness.

Reply:
I was wrong: P.DG identifies the board to, but I'm pretty sure it's not needed to univocally identify the PCB, so I would not add it for shortnessa

Reply:
I look at this from a different angle: Many problems with device names are created because we try to think how we could make names shorter and what we think will create unique names at the present time. Then, some new unanticipated device comes up and our great plan of uniqueness is destroyed.
Thanks why I initially motivated you to take the full name for the device string as well, as this is the easiest and cleanest approach, with the only disadvantage of having long device names.
Despite, consistency between model/DEVICE_MODEL and the image name will also help users looking for a device. Thus, I would still prefer to add the pdg after all. When I google for "A4001N", the P.DG is always there, so it's part of the name after all.

Reply:
Ok, so you want me to add "pdg" (or "p-dg" ?) to all the file names in the sources?

Reply:
I think the only file name would be the DTS, but I just think of
adb_p-dg-a4001n-a-000-1a1-ax
or
adb_pdg-a4001n-a-000-1a1-ax
(or with comma for the compatible). Won't make it much worse btw.

The only thing I'm not sure about yet is the led labels, but if I stick to my own concept they also need the long name (without vendor).

Reply:
ok, so you want me to change the DTS name only? I think I will use "adb_pdg-a4001n-a-000-1a1-ax".

The real problem with LEDs, in my opinion, is that their long name "A4001N_A-000-1A1-AX:green:internet" is shown il LUCI too. To me it has not to be shown. It should be just "green:internet"

Reply:
I want to have compatible, model name, DEVICE_MODEL, device node name, DTS file name, and LED model prefix with the same "name", e.g. with pdg in this case.
The LED thing is very annoying indeed, unfortunately having the model there seems to be a (though only formal) kernel requirement, I also do not see a practical use in it.

I'd prefer lower-case and hyphens only for the LED prefix, that's typical at other more modern targets, as it allows to generate the prefix from device node name:
adb_pdg-a4001n-a-000-1a1-ax -> "cut at underscore" -> pdg-a4001n-a-000-1a1-ax

Then, you do not have to put the long board names in base-files explicitly, but can just use a variable, e.g. https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/generic/base-files/etc/board.d/01_leds#L12